### DISCUSSION TITLE
Dynamic model loading in modal windows, lazy/defer load invisible components ?

### DISCUSSION THREAD

**Discussion by rstanislav** [livewire/livewire#4039] [1. Help]
Hello! I have livewire component that are used for modal windows inside other component, at page load its invisible, in blade i have:

`@livewire('one-order-add-change-client', ['order' => $order])`

Inside that component i have isVisible attribute, that is false on page load, when user clicks button, i emit event that catched inside that component and changes isVisible to true, its is working ok, BUT main page in very complex, alot of data and other components etc.. and as result even with most components invisible and not rendered at start they all load data from DB, so i have over 700 models loaded on average, and when they change state to visible they again reload data that belongs to them (in case one-order-add-change-client its about 200 models).
 What i'm doing wrong and how to prevent that ? For initial page load i only need about ~ 150 models because all other info is invisible and not even rendered in DOM.

**Comment by walirazzaq** [comment]
I am not sure but from you description I am assuming that you it might be taking a network round trip to just fire the `event` to `modal` component. 
If you are just concerned front end event and use `Livewire.emit(..)` or `Livewire.emitTo('component-name',...)`  .You can avoid the round trip to parent component and emit event staying on frontend side. Further if you show the code it could better serve the problem

You can also check [livewire-ui/modal](https://github.com/livewire-ui/modal) and awesome package to tackle modals in livewire
